src/gui/qgsadvanceddigitizingdockwidget.h

```diff
@@ -368,6 +370,77 @@ class GUI_EXPORT QgsAdvancedDigitizingDockWidget : public QgsDockWidget, private
      */
     void pointChanged( const QgsPointXY &point );
 
+    //! Signals for external widgets that need to update according to current values
+
+    /**
+    * Emitted whenever the value changes (either the mouse moved, or the user changed the input).
+    * Could be used by widgets that must reflect the current advanced digitizing state.
+    *
+    * \param value The current value (as a QString, as it could be an expression).
+    */
+    void valueXChanged( QString value );
+    void valueYChanged( QString value );
+    void valueAngleChanged( QString value );
+    void valueDistanceChanged( QString value );
+
+    /**
+    * Emitted whenever the parameter is locked.
+    * Could be used by widgets that must reflect the current advanced digitizing state.
+    *
+    * \param locked Whether the parameter is locked or not.
+    */
+    void lockXChanged( bool locked );
+    void lockYChanged( bool locked );
+    void lockAngleChanged( bool locked );
+    void lockDistanceChanged( bool locked );
+
+    /**
+    * Emitted whenever the parameter is toggled between absolute and relative.
+    * Could be used by widgets that must reflect the current advanced digitizing state.
+    *
+    * \param relative Whether the parameter is relative or not.
+    */
+    void relativeXChanged( bool relative );
+    void relativeYChanged( bool relative );
+    void relativeAngleChanged( bool relative );
+    // void relativeDistanceChanged( bool relative ); // distance is always relative
+
+    /**
+    * Emitted whenever the field is enabled or disabled. Depending on the context, some parameters
+    * dont make sense (e.g. you need a previous point to define a distance).
+    * Could be used by widgets that must reflect the current advanced digitizing state.
+    *
+    * \param enabled Whether the parameter is enabled or not.
+    */
+    void enabledChangedX( bool enabled );
+    void enabledChangedY( bool enabled );
+    void enabledChangedAngle( bool enabled );
+    void enabledChangedDistance( bool enabled );
+
+    /**
+    * Emitted whenever the field should get the focus using the shortcuts (X, Y, A, D).
+    * Could be used by widgets to capture the focus when a field is being edited.
+    */
+    void focusOnX();
+    void focusOnY();
+    void focusOnAngle();
+    void focusOnDistance();
+
+
+  public slots:
+
+    /**
+    * Set and lock the parameter.
+    * Can be used to set constraints.
+    *
+    * \param value The current value (as a QString, as it could be an expression).
+    */
+    void setX( QString value );
```

Comment:
I think those are all normal public functions and not used as slots (always called from lambdas).

Reply:
I thought that sometimes it may be interesting for widgets to use them as slots (e.g. to connect a QEditLine::textChanged). Is there a downside to have them as slots ? Or would it be possible to later on change them as slots without breaking backwards compatilibility ?

Reply:
The only downside is, that it adds a bit of overhead, so no big deal.
But on the other hand, the advantage is tiny as well: somebody might need a different signature (e.g. double as parameter), and it's very easy to trigger them with lambdas as you did.
Upgrading it to a slot when needed in the future will be trivial. Downgrading not (because it will be stable API)

Reply:
Ok, changed to normal functions